inbm-lib/inbm_common_lib/utility.py

```diff
@@ -205,7 +205,9 @@ def is_within_directory(directory: str, target: str) -> bool:
     return prefix == abs_directory
 
 
-def safe_extract(tarball: tarfile.TarFile, path: str = ".", members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
+def safe_extract(tarball: tarfile.TarFile,
+                 path: str = ".",
+                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
```

Comment:
**Suggestion:** Refactor the safe_extract function to include a check for member paths that are absolute or escape the extraction directory, which can be a security risk. [security]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
def safe_extract(tarball: tarfile.TarFile,
                 path: str = ".",
                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
    """Avoid path traversal when extracting tarball
    @param tarball: tarball to extract
    @param path: path to extract to
    @param members: members to extract
    @param numeric_owner: whether to extract numeric owner
    """
    if members is None:
        members = tarball.getmembers()
    for member in members:
        member_path = os.path.join(path, member.name)
        if os.path.isabs(member.name) or not is_within_directory(path, member_path):
            raise IOError(f"Unsafe path detected: {member.name}")
    tarball.extractall(path, members, numeric_owner=numeric_owner)
```